examples/submit-guppy-h1-h2-via-qir.ipynb

```diff
@@ -216,23 +235,346 @@
+      "  tail call void @__quantum__qis__mz__body(%Qubit* nonnull inttoptr (i64 1 to %Qubit*), %Result* nonnull inttoptr (i64 1 to %Result*))\n",
+      "  %3 = tail call i1 @__quantum__qis__read_result__body(%Result* nonnull inttoptr (i64 1 to %Result*))\n",
+      "  br i1 %3, label %4, label %cond_104_case_0\n",
+      "\n",
+      "4:                                                ; preds = %cond_exit_176\n",
+      "  %5 = add nsw i64 %\"15_0.0221\", 1\n",
+      "  tail call void @__quantum__qis__phasedx__body(double 0x3FF921FB54442D18, double 0xBFF921FB54442D18, %Qubit* null)\n",
+      "  tail call void @__quantum__qis__rz__body(double 0x400921FB54442D18, %Qubit* null)\n",
+      "  br label %cond_104_case_0\n",
+      "}\n",
+      "\n",
+      "declare void @__quantum__qis__phasedx__body(double, double, %Qubit*) local_unnamed_addr\n",
+      "\n",
+      "declare void @__quantum__qis__rz__body(double, %Qubit*) local_unnamed_addr\n",
+      "\n",
+      "declare void @__quantum__qis__mz__body(%Qubit*, %Result*) local_unnamed_addr\n",
+      "\n",
+      "declare i1 @__quantum__qis__read_result__body(%Result*) local_unnamed_addr\n",
+      "\n",
+      "declare void @__quantum__rt__bool_record_output(i1, i8*) local_unnamed_addr\n",
+      "\n",
+      "attributes #0 = { \"entry_point\" \"output_labeling_schema\" \"qir_profiles\"=\"custom\" \"required_num_qubits\"=\"3\" \"required_num_results\"=\"3\" }\n",
+      "\n",
+      "!llvm.module.flags = !{!0, !1, !2, !3}\n",
+      "\n",
+      "!0 = !{i32 1, !\"qir_major_version\", i32 1}\n",
+      "!1 = !{i32 7, !\"qir_minor_version\", i32 0}\n",
+      "!2 = !{i32 1, !\"dynamic_qubit_management\", i1 false}\n",
+      "!3 = !{i32 1, !\"dynamic_result_management\", i1 false}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "guppy_qir = hugr_to_qir(main.compile(), emit_text=True, validate_qir=False)\n",
+    "print(guppy_qir)"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "8b682b22-fd52-4b39-8c81-20939880862d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Validation failed as expected:\n",
+      "Found loop in CFG containing the block: cond_exit_176\n"
+     ]
+    }
+   ],
+   "source": [
+    "# this will fail because of the loop in the generated QIR\n",
+    "\n",
+    "try:\n",
+    "    guppy_qir = hugr_to_qir(main.compile(), emit_text=True, validate_qir=True)\n",
+    "    print(guppy_qir)\n",
+    "except Exception as e:\n",
+    "    print('Validation failed as expected:')\n",
+    "    print(e)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "bc85bba4-9e7b-48ee-b2e5-d046ad6108ad",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "278320c9-ed96-4556-af12-f7522b938d27",
+   "id": "7855cb3c-1bb0-457c-bd3a-4d16901fe4d6",
```

Comment:
There are some empty cells at the bottom of this notebook which can probably be removed.

Reply:
done, see [5fefa66](https://github.com/CQCL/hugr-qir/pull/81/commits/5fefa6686f019be610aee85d612bf8e516031b38)